web3/utils/events.py

```diff
@@ -238,3 +254,206 @@ def remove_trailing_from_seq(seq, remove_value=None):
 normalize_topic_list = compose(
     remove_trailing_from_seq(remove_value=None),
     pop_singlets,)
+
+
+def is_indexed(arg):
+    if isinstance(arg, TopicArgumentFilter) is True:
+        return True
+    return False
+
+
+is_not_indexed = complement(is_indexed)
+
+
+class Flag:
+    _flag = None
+
+    def __bool__(self):
+        if self._flag is None or self._flag is False:
+            return False
+        else:
+            return True
+
+    def __call__(self, flag):
+        self._flag = flag
+
+
+class EventFilterBuilder:
+    formatter = None
+    _fromBlock = None
+    _toBlock = None
+    _address = None
+
+    def __init__(self, event_abi, formatter=None):
+        self._deployed_flag = Flag()
+        self.event_abi = event_abi
+        self.formatter = formatter
+        self.event_topic = initialize_event_topics(self.event_abi)
+        self.args = AttributeDict(
+            _build_argument_filters_from_event_abi(event_abi, self._deployed_flag))
```

Comment:
Re:

>  Is this an anti-pattern? It feels like an anti-pattern.

Yeah, I think so too. Maybe instead of this shared flag, replace the `args` attribute in `deploy()`. Something like (pseudocode):

```diff
-        self._deployed_flag(True)
+        old_args = self.args
+        self.args = ImmutableArgs(old_args)
+        old_args.clear()  # make it clear that people should not use an old saved copy of args
```